src/indexing.jl

```diff
@@ -124,15 +129,20 @@ axisindexes(ax, idx) = axisindexes(axistrait(ax.val), ax.val, idx)
 axisindexes(::Type{Unsupported}, ax, idx) = error("elementwise indexing is not supported for axes of type $(typeof(ax))")
 axisindexes(t, ax, idx) = error("cannot index $(typeof(ax)) with $(typeof(idx)); expected $(eltype(ax)) axis value or Integer index")
 
-# Dimensional axes may be indexed directy by their elements if Non-Real and unique
+# Dimensional axes may be indexed directly by their elements if Non-Real and unique
 # Maybe extend error message to all <: Numbers if Base allows it?
-axisindexes{T<:Real}(::Type{Dimensional}, ax::AbstractVector{T}, idx::T) = error("indexing by axis value is not supported for axes with $(eltype(ax)) elements; use an ClosedInterval instead")
+axisindexes{T<:Real}(::Type{Dimensional}, ax::AbstractVector, idx::T) =
+    throw(ArgumentError("invalid index: $idx. Use `atvalue` when indexing by value."))
 function axisindexes(::Type{Dimensional}, ax::AbstractVector, idx)
     idxs = searchsorted(ax, ClosedInterval(idx,idx))
     length(idxs) > 1 && error("more than one datapoint lies on axis value $idx; use an interval to return all values")
     idxs[1]
 end
 
+# Dimensional axes may always be indexed by value if in a Value type wrapper.
+axisindexes(::Type{Dimensional}, ax::AbstractVector, idx::Value) =
+    findfirst(ax.==idx.val)
```

Comment:
This implementation really could be the same as the method on [line 136](https://github.com/JuliaArrays/AxisArrays.jl/pull/52#discussion-diff-86356474R136).  I'm not sure why I spelled it that way, but I think it had something to do with some types not implementing the comparisons that I needed.